infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -128,32 +146,77 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      *
      * @param txnContext All of the above must execute in the same transaction passed in.
      */
-    public void migrateData(TxnContext txnContext) {
+    public void migrateData(CorfuStore corfuStore, TxnContext txnContext) {
         // Currently only the LogReplicationMetadataManager needs data-migration
         LogReplicationMetadataManager.migrateData(txnContext);
-        addSnapshotSyncEventOnUpgradeCompletion(txnContext);
+        addSnapshotSyncEventOnUpgradeCompletion(corfuStore, txnContext);
     }
 
     /**
-     * Add flag to event table to trigger snapshot sync.
+     * This is a helper method that is used to construct the sessions from the existing V1 metadata
+     * instead of otherwise relying on the session manager. Intended for FULL_TABLE replication.
      */
-    public void addSnapshotSyncEventOnUpgradeCompletion(TxnContext txnContext) {
-        UUID rollingUpgradeForceSyncId = UUID.randomUUID();
-
-        // Write a rolling upgrade force snapshot sync event to the logReplicationEventTable
-        LogReplicationMetadata.ReplicationEventInfoKey key = LogReplicationMetadata.ReplicationEventInfoKey.newBuilder()
-                .build();
-
-        LogReplicationMetadata.ReplicationEvent event = LogReplicationMetadata.ReplicationEvent.newBuilder()
-                .setEventId(rollingUpgradeForceSyncId.toString())
-                .setType(LogReplicationMetadata.ReplicationEvent.ReplicationEventType.UPGRADE_COMPLETION_FORCE_SNAPSHOT_SYNC)
-                .setEventTimestamp(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build())
-                .build();
+    @VisibleForTesting
+    public List<LogReplicationSession> buildSessionsFromOldMetadata(CorfuStore corfuStore, TxnContext txnContext) {
+        List<LogReplicationSession> sessions = new ArrayList<>();
+
+        // Get the sinkClusterIds from the keys of the status table entries
+        List<ReplicationStatusKey> statusTableKeys = new ArrayList<>(txnContext.keySet(V1_STATUS_TABLE_NAME));
+        List<String> sinkClusterIds = statusTableKeys.stream()
+                .map(ReplicationStatusKey::getClusterId)
+                .collect(Collectors.toList());
+
+        if (sinkClusterIds.isEmpty()) {
+            log.info("No V1 metadata found");
+        } else {
+            // Get the localClusterIds from suffix's of the old metadata table names and filter out the
+            // known sink clusters IDs from the status table to get the sourceClusterIds
+            List<String> sourceClusterIds = corfuStore.listTables(CORFU_SYSTEM_NAMESPACE)
+                    .stream()
+                    .map(TableName::getTableName)
+                    .filter(fullName -> fullName.startsWith(V1_METADATA_TABLE_PREFIX))
+                    .map(fullName -> fullName.substring(V1_METADATA_TABLE_PREFIX.length()))
+                    .filter(remoteId -> !sinkClusterIds.contains(remoteId))
```

Comment:
does this filter exist because the sink cluster uses its own cluster id as the key in the status table?  It will be good to add a comment explaining it if so